odxtools/diagcodedtypes.py

```diff
@@ -315,6 +315,8 @@ def convert_internal_to_bytes(self, internal_value, encode_state: EncodeState, b
             return value_byte
         else:
             termination_char = self.__termination_character()
+            if self.termination == 'END-OF-PDU':
+                termination_char = bytes()
```

Comment:
Do you have an example where this is needed?

I thought a parameter with `termination == 'END-OF-PDU'` must be the last parameter in a request. In this case the flag `encode_state.is_end_of_pdu` should be True and this else-branch is never reached.
In this case, I would prefer to raise an error here, as this shouldn't happen..
(It's entirely possible that I misunderstand the usage of `MinMaxLengthType`s though)

Reply:
I don't know what exactly happened, but I do know that with the pdx I had the parser reached this point with `termination == 'END-OF-PDU'` and `termination_char == None`.

Reply:
Hm, ok, I would be interested in an example (ideally a unit test case) because I can't reproduce it..
That shouldn't stop this from being merged though.